index.js

```diff
@@ -1,183 +1,187 @@
+      pool = this.pools[database] = makePool(this.options, database);
     }
 
-    var self = this;
-    setInterval(function() {
-        Object.keys(self.pools).forEach(function(poolKey) {
-            var pool = self.pools[poolKey];
-            self.emit('status', {
-                name: self.options.name,
-                db: poolKey,
-                count: pool.getPoolSize(),
-                unused: pool.availableObjectsCount(),
-                waiting:  pool.waitingClientsCount()
-            });
-        });
-    }, this.options.emitter.statusInterval);
-}
-
-util.inherits(RedisPool, EventEmitter);
-
-
-module.exports = RedisPool;
+    const startTime = Date.now();
+    const client = await pool.acquire()
+    const elapsedTime = Date.now() - startTime;
 
-
-/**
- * Acquire resource
- *
- * @param {String|Number} database redis database name
- * @param {Function} callback Callback to call once acquired. Takes the form `callback(err, resource)`
- */
-RedisPool.prototype.acquire = function(database, callback) {
-    var self = this;
-    var pool = this.pools[database];
-    if (!pool) {
-        pool = this.pools[database] = makePool(this.options, database);
+    if (elapsedTime > this.options.slowPool.elapsedThreshold) {
+      log(this.options, { db: database, action: 'acquire', elapsed: elapsedTime, waiting: pool.pending });
     }
-    var startTime = Date.now();
-    pool.acquire(function(err, client) {
-        var elapsedTime = Date.now() - startTime;
-        if (elapsedTime > self.elapsedThreshold) {
-            log(self.options, {db: database, action: 'acquire', elapsed: elapsedTime, waiting: pool.waitingClientsCount()});
-        }
-        callback(err, client);
-    });
-};
 
+    return client;
+  }
 
-/**
- * Release resource.
- *
- * @param database {String|Number} redis database name
- * @param resource {Object} resource object to release
- */
-RedisPool.prototype.release = function(database, resource) {
+  /**
+   * Release resource.
+   *
+   * @param {String|Number} database redis database name
+   * @param {Object} resource resource object to release
+   */
+  async release (database, resource) {
     if (this.options.unwatchOnRelease) {
-        resource.UNWATCH();
+      resource.UNWATCH();
     }
 
-    var pool = this.pools[database];
+    const pool = this.pools[database];
 
     if (pool) {
-        pool.release(resource);
+      await pool.release(resource);
     }
-};
+  }
 
+  _addCommands () {
+    if (this.options.commands.length) {
+      this.options.commands.forEach(newCommand => redis.add_command(newCommand));
+    }
+  }
+
+  _emitStatus() {
+    setInterval(() => {
+      for (const [poolKey, pool] of Object.entries(this.pools)) {
+        this.emit('status', {
+          name: this.options.name,
+          db: poolKey,
+          count: pool.size,
+          unused: pool.available,
+          waiting: pool.pending
+        });
+      }
+    }, this.options.emitter.statusInterval);
```

Comment:
Ensure `this.options.emitter.statusInterval` is never `null` or `undefined`. It might flood the event loop with an infinite number of callbacks of this interval.

Reply:
the user should set is as null, but yes, it could happen. I am going to change it